Lib/cgi.py

```diff
@@ -589,13 +593,13 @@ def read_urlencoded(self):
         query = urllib.parse.parse_qsl(
             qs, self.keep_blank_values, self.strict_parsing,
             encoding=self.encoding, errors=self.errors,
-            max_num_fields=self.max_num_fields)
+            max_num_fields=self.max_num_fields, separator=separator)
```

Comment:
If you get the value from self here, then we won't need to pass in as an argument to ``read_urlencoded`` ;)
```suggestion
            max_num_fields=self.max_num_fields, separator=self.separator)
```